### PR TITLE
feat(buckets): store pagination and filter state in url on change

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes https://github.com/influxdata/ui/issues/2725

This PR was inspired from https://github.com/influxdata/ui/pull/2610, basically does the same thing as implemented for paginated tasks, for buckets.

Video: 

https://user-images.githubusercontent.com/18511823/135359863-cc623bcc-c708-4b43-82a7-1d5a68f98abd.mov

### CODE REVIEW

File: src/buckets/pagination/BucketList.tsx [hunk 1]
```diff
@@ -45,6 +45,18 @@ class BucketList
   public rowsPerPage: number = 10
   public totalPages: number
 
+  public componentDidMount() {
+    const params = new URLSearchParams(window.location.search)
+    const urlPageNumber = parseInt(params.get('page'), 10)
+
+    const passedInPageIsValid =
+      urlPageNumber && urlPageNumber <= this.totalPages && urlPageNumber >= 0
```
Comment (hoorayimhelping) [inline src/buckets/pagination/BucketList.tsx:53]: I think we want to change `urlPageNumber >=0` to `urlPageNumber > 0` - I think when I first coded this up, I wasn't offsetting the 0 index to 1. Can you play with this and see if 0 or 1 makes a little more sense?

**Review by hoorayimhelping — APPROVED** [review]
Solid work, you must have had some amazingly good code to base this off of  😆 😆 

One comment about the minimum page being 0 - could you poke around and see if 1 makes more sense?

File: src/buckets/pagination/BucketList.tsx (hunk 1 above)
Comment (ChitlangeSahas) [reply src/buckets/pagination/BucketList.tsx]: Just changed and tested, it works!

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2742)

**Pull request merged**